apps/examples/testcase/le_tc/filesystem/fs_main.c

```diff
@@ -2445,12 +2445,51 @@ static void tc_libc_stdio_setvbuf(void)
 	char buffer[64];
 	char *filename = VFS_FILE_PATH;
 	int ret;
+	int fp_fd;
 
 	/* setvbuf_test: DEFAULT buffering */
 
 	fp = fopen(filename, "w");
 	TC_ASSERT_NEQ("fopen", fp, NULL);
 
+	/* setvbuf_test: fail to file open */
+
+	fp_fd = fp->fs_fd;
+	fp->fs_fd = ERROR;
+	ret = setvbuf(fp, buffer, _IOFBF, 64);
+	fp->fs_fd = fp_fd;
+	TC_ASSERT_LEQ_CLEANUP("setvbuf", ret, 0, fclose(fp));
+	TC_ASSERT_EQ_CLEANUP("setvbuf", *get_errno_ptr(), EBADF, fclose(fp));
```

Comment:
You should use `errno` instead of `*get_errno_ptr()` for protected build.

Reply:
modify